apis/tempo/v1alpha1/tempomonolithic_types.go

```diff
@@ -0,0 +1,244 @@
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TempoMonolithicSpec defines the desired state of TempoMonolithic.
+type TempoMonolithicSpec struct {
+	// Storage defines the backend storage configuration
+	//
+	// +kubebuilder:validation:Optional
+	Storage *MonolithicStorageSpec `json:"storage,omitempty"`
+
+	// Ingestion defines the trace ingestion configuration
+	//
+	// +kubebuilder:validation:Optional
+	Ingestion *MonolithicIngestionSpec `json:"ingestion,omitempty"`
+
+	// JaegerUI defines the Jaeger UI configuration
+	//
+	// +kubebuilder:validation:Optional
+	JaegerUI *MonolithicJaegerUISpec `json:"jaegerui,omitempty"`
+
+	// ManagementState defines whether this instance is managed by the operator or self-managed
+	//
+	// +kubebuilder:validation:Optional
+	Management ManagementStateType `json:"management,omitempty"`
+
+	// Observability defines observability configuration for the Tempo deployment
+	//
+	// +kubebuilder:validation:Optional
+	Observability *MonolithicObservabilitySpec `json:"observability,omitempty"`
+
+	// ExtraConfig defines any extra (overlay) configuration for components
+	//
+	// +kubebuilder:validation:Optional
+	ExtraConfig *MonolithicExtraConfigSpec `json:"extraConfig,omitempty"`
```

Comment:
Any reason for not reusing the same as microservices? The only reason I think is because the microservices could include other configs in the future. If that is the reason I'm ok

Reply:
I've updated the PR and reused the struct and logic from the TempoStack now.